### PR TITLE
added functionality to customize the MG scope

### PR DESCRIPTION
**Pull request by AmitSheth**
I am deploying this to a client that did not want the Tool / App Registrations to have access at the Tenant Root Group. Instead they wanted it scoped to a lower Management Group. 

Added a parameter to the deployment script to provide the Management Group ID. Default is still Tenant Root Group.
Updated documentation.
Tested in my sandbox and works correctly.

### CODE REVIEW

File: docs/deployment/README.md [hunk 1]
```diff
@@ -136,6 +136,14 @@ You have the ability to pass optional flags to the deployment script:
 
 > **NOTE 3:** Maximum of seven (7) characters. This is because the prefix is used to generate names for several different Azure resource types with varying maximum lengths.
 
+**Customize the Management Group that the App Registrions have access to. Default is Tenant Root Group:**
```
Comment (DCMattyG) [inline docs/deployment/README.md:139]: Please check your spelling here (e.g. "Registrions")

File: docs/how-to/README.md [hunk 2]
```diff
@@ -6,7 +6,7 @@
...
+IPAM has the concept of an **IPAM Administrator**. While using the IPAM tool as an administrator, you are viewing Azure resources through the permissions of the Engine Service Principal, which has [Reader](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#reader) at the [Tenant Root Management Group](https://learn.microsoft.com/en-us/azure/governance/management-groups/overview#root-management-group-for-each-directory) level or a Custom Management Group. Upon initial deployment, no IPAM administrators are set which has the effect of **all** users having administrative rights. You can define who within your Azure AD Tenant should be designated as an IPAM administrator via the **Admin** section of the menu blade.
```
Comment (DCMattyG) [inline docs/how-to/README.md:9]: This should be explained better, such as:

"...which, by default, has Reader at the Tenant Root Group, or alternatively a Management Group of your choice."

File: docs/deployment/README.md [hunk 3]
```diff
@@ -136,6 +136,14 @@ You have the ability to pass optional flags to the deployment script:
...
+
+```powershell
+./deploy.ps1 `
+  -ManagementGroupID "my-custom-management-group"
```
Comment (DCMattyG) [inline docs/deployment/README.md:143]: You need to add the "-ManagementGroupID" flag into the table above, it shouldn't only appear in the examples.

File: docs/deployment/README.md [hunk 4]
```diff
@@ -136,6 +136,14 @@ You have the ability to pass optional flags to the deployment script:
...
+  -Location "westus3" `
+```
+
```
Comment (DCMattyG) [inline docs/deployment/README.md:146]: This section only appears in the "Full" deployment docs/examples...but it also applies to the "Infrastructure Only" deployment method as well.

File: deploy/deploy.ps1 [hunk 5]
```diff
@@ -778,11 +790,6 @@ process {
     }
 
     if ($PSCmdlet.ParameterSetName -in ('Full', 'AppsOnly', 'Function', 'FuncAppsOnly')) {
-      # Fetch Tenant ID
```
Comment (DCMattyG) [inline deploy/deploy.ps1:781]: Please add this back and make it the "default" if no "ManagementGroupID" was explicitly passed into the Deployment script.

File: deploy/deploy.ps1 [hunk 6]
```diff
@@ -807,7 +814,7 @@ process {
       $appDetails = Deploy-IPAMApplications `
         -UIAppName $UIAppName `
         -EngineAppName $EngineAppName `
-        -TenantId $tenantId `
+        -ManagementGroupID $ManagementGroupID `
```
Comment (DCMattyG) [inline deploy/deploy.ps1:817]: The default should be passed here, ideally.

File: deploy/deploy.ps1 (hunk 5 above)
Comment (AmitSheth) [reply deploy/deploy.ps1:781]: Added Back

File: deploy/deploy.ps1 (hunk 6 above)
Comment (AmitSheth) [reply deploy/deploy.ps1:829]: Default set in code above if Management Group ID is not explicitly set.

File: docs/deployment/README.md (hunk 3 above)
Comment (AmitSheth) [reply docs/deployment/README.md:146]: Added

File: docs/deployment/README.md (hunk 4 above)
Comment (AmitSheth) [reply docs/deployment/README.md:149]: I think you meant to say apps only. Added there as well.

File: deploy/deploy.ps1 [hunk 7]
```diff
@@ -12,6 +12,18 @@
...
+  Mandatory = $true,
+  ParameterSetName = 'Full')]
+  [Parameter(ValueFromPipelineByPropertyName = $true,
+    Mandatory = $true,
+    ParameterSetName = 'TemplateOnly')]
+  [Parameter(ValueFromPipelineByPropertyName = $true,
+    Mandatory = $true,
+    ParameterSetName = 'Function')]
+  [string]
+  $ManagementGroupID = $((Get-AzContext).Tenant.Id),
```
[the remark this answers is not in the document]
Comment (AmitSheth) [reply deploy/deploy.ps1]: Understood. Moved the logic.

File: docs/deployment/README.md (hunk 1 above)
Comment (AmitSheth) [reply docs/deployment/README.md]: Fixed.

File: docs/how-to/README.md (hunk 2 above)
Comment (AmitSheth) [reply docs/how-to/README.md]: Updated

File: docs/deployment/README.md (hunk 4 above)
Comment (DCMattyG) [reply docs/deployment/README.md:149]: I did indeed, thank you. Needed more coffee apparently 😆

**Comment by AmitSheth** [comment]
@DCMattyG  anything else I need to do on this?

**Comment by DCMattyG** [comment]
Good morning @AmitSheth, not right now. Thank you for all of the updates!

I'm just going to go through some testing on my end, then we should be good to go 😉

**Comment by bluedemise** [comment]
has this PR been solved elsewhere or just abandoned? We would still admire a merge of the PR.

**Comment by DCMattyG** [comment]
Hi @bluedemise, not abandoned but there were some other major changes I was focused on. I'll make sure this gets into the next release.

Thanks for the bump on this!

**Comment by bluedemise** [comment]
> Hi @bluedemise, not abandoned but there were some other major changes I was focused on. I'll make sure this gets into the next release.
> 
> Thanks for the bump on this!

Hi @DCMattyG
Is there a release on the horizon were this will get included?
Thanks for your work!

**Comment by DCMattyG** [comment]
Application & deployment script were redesigned for version 3.0. Incorporated these update in #279 and are now incorporated as of v3.2.0.

Thanks for the contributions, glad to finally have this option available once again for folks doing PoC/test deployments.

**Pull request closed without merging**